cli/command/service.go

```diff
@@ -10,6 +15,261 @@ func NewServiceCommand(b *CommandBase) *ServiceCommand {
+			},
+			{
+				Name:      "scale",
+				Usage:     "scale a service",
+				Action:    s.scale,
+				ArgsUsage: "NAME COUNT",
+				Flags: []cli.Flag{
+					cli.BoolFlag{
+						Name:  "nowait",
+						Usage: "don't wait for the job to complete before returning",
+					},
+				},
+			},
+			{
+				Name:      "update",
+				Usage:     "run a new dploy on a service",
+				Action:    s.update,
+				ArgsUsage: "NAME DEPLOY",
+				Flags: []cli.Flag{
+					cli.BoolFlag{
+						Name:  "nowait",
+						Usage: "don't wait for the job to complete before returning",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (s *ServiceCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "NAME", "DEPLOY")
+	if err != nil {
+		return err
+	}
+
+	deployID, err := s.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := s.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	var loadBalancerID string
+	if loadBalancerName := c.String("loadbalancer"); loadBalancerName != "" {
+		id, err := s.resolveSingleEntityIDHelper("load_balancer", loadBalancerName)
+		if err != nil {
+			return err
+		}
+
+		loadBalancerID = id
+	}
+
+	req := models.CreateServiceRequest{
+		DeployID:       deployID,
+		EnvironmentID:  environmentID,
+		LoadBalancerID: loadBalancerID,
+		ServiceName:    args["NAME"],
+	}
+
+	jobID, err := s.client.CreateService(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFn := func(serviceID string) error {
+		service, err := s.client.ReadService(serviceID)
+		if err != nil {
+			return err
+		}
+
+		return s.printer.PrintServices(service)
+	}
+
+	return s.waitOnJobHelper(c, jobID, "creating", onCompleteFn)
+}
+
+func (s *ServiceCommand) delete(c *cli.Context) error {
+	deleteFn := func(serviceID string) (string, error) {
+		return s.client.DeleteService(serviceID)
+	}
+
+	return s.deleteHelper(c, "service", deleteFn)
+}
+
+func (s *ServiceCommand) list(c *cli.Context) error {
+	serviceSummaries, err := s.client.ListServices()
+	if err != nil {
+		return err
+	}
+
+	return s.printer.PrintServiceSummaries(serviceSummaries...)
+}
+
+func (s *ServiceCommand) logs(c *cli.Context) error {
+	return nil
+}
+
+func (s *ServiceCommand) read(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME")
+	if err != nil {
+		return err
+	}
+
+	serviceID, err := s.resolveSingleEntityIDHelper("service", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	service, err := s.client.ReadService(serviceID)
+	if err != nil {
+		return err
+	}
+
+	return s.printer.PrintServices(service)
+}
+
+func (s *ServiceCommand) scale(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME", "COUNT")
+	if err != nil {
+		return err
+	}
+
+	serviceID, err := s.resolveSingleEntityIDHelper("service", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	scale, err := strconv.Atoi(args["COUNT"])
+	if err != nil {
+		return err
```

Comment:
we should probably wrap this error with something like `fmt.Errorf("Failed to parse COUNT argument: %v", err)`

Reply:
✅